.github/workflows/test_coverage.yml

```diff
@@ -27,6 +27,7 @@ jobs:
         pip install pytest
         pip install pytest-mock
         pip install posix-ipc
+        pip install numpy
```

Comment:
We shouldn't have this libraries installed in the test environment but I will approve it for now. just add a todo note please